src/pages/product-listing/ProductListing.jsx

```diff
@@ -0,0 +1,64 @@
+import { ContainerCard, Filter, Navigation } from "../../components";
+import "./produc-listing.css";
+import React, { useEffect } from "react";
+import axios from "axios";
+import { useFilter, useProduct } from "../../context";
+
+export default function ProductListing() {
+  const {
+    FilterState: { byCategory, sortBy, byRating, byPrice },
+        
+  } = useFilter();
+  const { ProductState, ProductDispatch } = useProduct();
+
+  const transformProduct = () => {
+    let sortedProduct = ProductState.productdata;
+
+    if (sortBy) {
+      sortedProduct = sortedProduct.sort((a, b) =>
+        sortBy === "PRICE_LOW_TO_HIGH" ? a.price - b.price : b.price - a.price
+      );
+    }
+    if (byCategory) {
+      sortedProduct = sortedProduct.filter(
+        (prod) => prod.categoryName === byCategory
+      );
+    }
+    if (byRating) {
+      sortedProduct = sortedProduct.filter((prod) => prod.rating >= byRating);
+    }
+    if (byPrice) {
+      sortedProduct = sortedProduct.filter((prod) => prod.price <= byPrice);
+    }
+    return sortedProduct;
+  };
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get("/api/products");
+        ProductDispatch({ type: "savedata", payload: response.data.products });
+      } catch (error) {
+        console.log(error, "Could not load data");
```

Comment:
Intead of logging the error you could add an alert here, it would be a better user experience